subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/LocationsWrittenByCurrentBuild.java

```diff
@@ -17,14 +17,22 @@
 package org.gradle.internal.watch.vfs.impl;
 
 import org.gradle.internal.file.FileHierarchySet;
+import org.gradle.internal.snapshot.CaseSensitivity;
 import org.gradle.internal.vfs.FileSystemAccess;
 
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 
 public class LocationsWrittenByCurrentBuild implements FileSystemAccess.WriteListener {
     private final AtomicReference<FileHierarchySet> producedByCurrentBuild = new AtomicReference<>(FileHierarchySet.empty());
+    private final AtomicReference<VersionHierarchy> versionHierarchy;
+    private final AtomicLong currentVersion = new AtomicLong(0);
     private volatile boolean buildRunning;
 
+    public LocationsWrittenByCurrentBuild(CaseSensitivity caseSensitivity) {
+        this.versionHierarchy = new AtomicReference<>(VersionHierarchy.empty(currentVersion.incrementAndGet(), caseSensitivity));
```

Comment:
To me it'd make sense if `VersionHierarchy.empty()` would initialize to `version = 0`, and we'd initialize `currentVersion` from `versionHierarchy.getVersion()`.